downsampler/src/main/scala/filodb.downsampler/PerSparkExecutorState.scala

```diff
@@ -0,0 +1,73 @@
+package filodb.downsampler
+
+import com.typesafe.scalalogging.StrictLogging
+import monix.execution.Scheduler
+
+import filodb.cassandra.columnstore.CassandraColumnStore
+import filodb.core.memstore.{TimeSeriesShardStats, WriteBufferPool}
+import filodb.core.metadata.{Dataset, Schemas}
+import filodb.memory._
+
+object PerSparkExecutorState extends StrictLogging {
+
+  import DownsamplerSettings._
+
+  val readSched = Scheduler.io("cass-read-sched")
+  val writeSched = Scheduler.io("cass-write-sched")
+  val cassandraColStore = new CassandraColumnStore(filodbConfig, readSched, None)(writeSched)
+
+  val kamonTags = Map( "rawDataset" -> rawDatasetName,
+                       "run" -> "Downsampler",
+                       "userTimeStart" -> userTimeStart.toString,
+                       "userTimeEnd" -> userTimeEnd.toString)
+
+  val schemas = Schemas.fromConfig(filodbConfig).get
+
+  val rawSchemas = rawSchemaNames.map { s => schemas.schemas(s)}
+  /**
+    * Downsample Schemas
+    */
+  val dsSchemas = rawSchemaNames.map { s => schemas.schemas(s).downsample.get}
+
+  /**
+    * Chunk Downsamplers by Raw Schema Id
+    */
+  val chunkDownsamplersByRawSchemaId = rawSchemas.map { s => s.schemaHash -> s.data.downsamplers }.toMap
```

Comment:
I suggest using a `debox.Map` here, or some other map that does not box integers.  Otherwise every map lookup will involve boxing.

Reply:
👍